### ISSUE TITLE
[Bug]: In DetailsList, GroupHeader's onRenderTitle does not receive id as prop (used as aria-labelledby in row that GroupHeader renders)

### ISSUE THREAD

**Issue by cabillin** [microsoft/fluentui#32229]
### Library

React / v8 (@fluentui/react)

### System Info

```shell
System:
    OS: Windows 11 10.0.26120
    CPU: (24) x64 AMD Ryzen Threadripper PRO 5945WX 12-Cores
    Memory: 30.93 GB / 63.86 GB
  Browsers:
    Edge: Chromium (127.0.2651.74)
    Internet Explorer: 11.0.26100.1
```

### Are you reporting an Accessibility issue?

yes

### Reproduction

https://codepen.io/KnetTravlr/pen/abgwZPb?editors=1010

## Actual Behavior
id prop is not available in IGroupHeaderProps (available in GroupHeader's onRenderTitle)
As a result, we cannot render our grid cell with the correct id for the parent row's aria-labelledby to be valid.
This is acceptable in the repro since the AT is able to infer the row's accessible name from its children's accessible names. However, we have a much more complicated use case where the AT is NOT inferring the row's accessible name (and we need to use the aria-labelledby on a specific sub-element).
![image](https://github.com/user-attachments/assets/d4286145-0eda-4c22-a239-c80be87165a1)

## Expected Behavior
id used in parent row's aria-labelledby is passed to GroupHeader's onRenderTitle, so we can achieve the same programmatic mapping as when there is no override.
![image](https://github.com/user-attachments/assets/bdc76be2-700e-44ec-bbd5-c66a7c5661eb)

### Requested priority

Normal

### Products/sites affected

Internal Microsoft website

### Are you willing to submit a PR to fix?

no

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

**Comment by spmonahan** [comment]
[The header id is generated here with no option for overriding](https://github.com/microsoft/fluentui/blob/master/packages/react/src/components/GroupedList/GroupHeader.base.tsx#L54).

Setting `ariaLabel` for the `group` may be a temporary workaround: https://github.com/microsoft/fluentui/blob/master/packages/react/src/components/GroupedList/GroupHeader.base.tsx#L130-L131
